rbi/core/dir.rbi

```diff
@@ -567,18 +567,10 @@ class Dir < Object
   # Equivalent to calling `Dir.glob([string,...], 0)`.
   sig do
     params(
-        pattern: T.any(String, T::Array[String]),
-        flags: Integer,
+        pattern: String,
```

Comment:
This should probably be `T.any(String, Pathname)` since, despite not being documented, Ruby makes an explicit attempt to convert non-String values by calling `FilePathValue` here: https://github.com/ruby/ruby/blob/1020e120e060c00eca456d5a129c344daa472407/dir.c#L2875

Reply:
How did you find the relationship between `Pathname` and `FilePathValue`?

It looks like `Pathname` needs an explicit require. Is it okay to add this to the rbi?

Reply:
It is a define in the Ruby codebase that calls the `rb_get_path` function: https://github.com/ruby/ruby/blob/ba7f4bb0af82bc9a7e392608fd7f6a316bec7c9f/include/ruby/ruby.h#L63

It is perfectly fine to use `Pathname` here, even though it is not available by default. There are similar usages in `File` methods too.